src/components/Button/Button.tsx

```diff
@@ -0,0 +1,18 @@
+import "../Button/Button.scss";
+import { MouseEventHandler } from "react";
+
+type ButtonProps = {
+  label: string;
+  variant: "default" | "secondary" | "custom";
```

Comment:
This is not how a custom style prop would really work.

Have a look at this -> [Style with props](https://lumin8media.com/blog/pass-style-as-prop-react-typescript)

Reach out if you have any questions.

Reply:
We have made the variant prop optional and improved the default. Is there anything else needed to fulfil the criteria?

Reply:
has been altered to use the cssproperties hook. and styles and such to match. Hopefully this is more what you were looking for.